### PR TITLE
Feat/3171 nexus integration

### LINKED ISSUE #3171: CircleCI pipeline utilizes internal Nexus infrastructure
**Description:**
To fully insulate our pipeline being broken by issues out of our control via upstream issues, we should ensure all steps can be self-sufficient within our infrastructure. For higher environments (HHS repo), we should ensure backwards compatibility with standard internet mirrors

**Acceptance Criteria:** 

- [x] Pipeline builds don't reach out to the internet by default
- [x] Pipeline builds will fail-through to internet repositories if Nexus connection fails
- [x] Pipeline build times are reduced
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

**Tasks:**

- [x] Apt repo configured
- [x] NPM repo configured
- [x] PyPI repo configured
- [x] Have CloudFoundry CLI be available in the pipeline through Nexus
- [x] Have alternative sources point to our standard internet repositories
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**
Implementation of a base image is out of scope, this is an infrastructure buildout.

**Supporting Documentation:**

**Open Questions:**

- How do we build once and reference the docker image moving forward? Out of scope, will generate another docker-nexus ticket.
- How do we correctly cache dependencies/images? Out of scope, will generate another docker-nexus ticket.
-

### PR DESCRIPTION
**Pull request by andrew-jameson**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #3171

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. All steps and success will be captured in pipelines.
Screenshot 2024-10-14 at 11 22 57 AM
Screenshot 2024-10-14 at 11 24 40 AM
Screenshot 2024-10-14 at 11 30 53 AM

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Pipeline builds don't reach out to the internet by default
- [x] Pipeline builds will fail-through to internet repositories if Nexus connection fails
- [x] Pipeline build times are reduced
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by elipe17 — APPROVED** [review]
Verified locally and pipelines look good!

**Review by jtimpe — APPROVED** [review]

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Comment by raftmsohani** [comment]
@elipe17 please resolve conflict before pushing to qasp

File: scripts/cf-check.sh [hunk 1]
```diff
@@ -3,15 +3,15 @@ set -e
...
     apt-get install wget gnupg2 apt-transport-https
 
-    wget -q -O - https://packages.cloudfoundry.org/debian/cli.cloudfoundry.org.key | sudo apt-key add -
-
-    echo "deb https://packages.cloudfoundry.org/debian stable main" | sudo tee /etc/apt/sources.list.d/cloudfoundry-cli.list
-
-    apt-get update
-    apt-get install cf7-cli
+    NEXUS_ARCHIVE="cf7-cli_7.7.13_linux_x86-64.tgz"
+    NEXUS_URL="https://tdp-nexus.dev.raftlabs.tech/repository/tdp-bin/cloudfoundry-cli/$NEXUS_ARCHIVE"
```
Comment (ADPennington) [inline scripts/cf-check.sh:10]: does this help keep our pipelines stable during the annual cf cli issue we've seen in recent years? @andrew-jameson @raftmsohani @jtimpe @elipe17

**Comment by ADPennington** [comment]
@andrew-jameson looks like there's a merge conflict here. is the pipeline running on this branch a sufficient test for this?

**Comment by andrew-jameson** [comment]
> @andrew-jameson looks like there's a merge conflict here. is the pipeline running on this branch a sufficient test for this?

Merge conflict resolved. Yes, pipeline is the only testing needed here as dev's have approved it works in their local environments.

File: scripts/cf-check.sh (hunk 1 above)
Comment (ADPennington) [reply scripts/cf-check.sh:10]: tagging one more time for visibility 😄 ⬆️  @andrew-jameson @jtimpe @raftmsohani @elipe17

Comment (elipe17) [reply scripts/cf-check.sh:10]: Yes indeed it should save us from them not rolling their certs in a timely manner.

**Review by ADPennington — APPROVED** [review]
LGTM 👍🏾

**Pull request merged**